src/mavis/summary/summary.py

```diff
@@ -190,43 +193,40 @@ def group_by_distance(calls, distances):
     return grouped_calls, removed_calls
 
 
-def annotate_dgv(bpps, dgv_regions_by_reference_name, distance=0):
+def annotate_dgv(
+    bpps: List[BreakpointPair],
+    dgv_regions_by_reference_name: Dict[Tuple[str], List[BreakpointPair]],
+    input_cluster_radius,
+):
     """
-    given a list of bpps and a dgv reference, annotate the events that are within the set distance of both breakpoints
+    Given a list of breakpoint pairs (bpps) and bpps from another reference set), annotate the events that are within the set distance of both breakpoints
 
     Args:
         bpps (list) : the list of BreakpointPair objects
-        dgv_regions_by_reference_name (dict) : the dgv reference regions file loaded by load_masking_regions
-        distance (int) : the minimum distance required to match a dgv event with a breakpoint
+        dgv_regions_by_reference_name (dict) : tuple of (break1_chr,break2_chr) and its associated list of BreakpointPair objects specified by the MAVIS input file
+        cluster_radius (int) : Distance used in matching input SVs to reference SVs through clusterind, defined by summary.cluster_radius in the configuration file
     """
-    for chrom in dgv_regions_by_reference_name:
-        dgv_regions_by_reference_name[chrom] = sorted(
-            dgv_regions_by_reference_name[chrom], key=lambda x: x.start
-        )
-
-    lowest_resolution = max([len(b.break1) for b in bpps])  # only need start res
-
-    # only look at the bpps that dgv events could pair to, Intrachromosomal
-    for bpp in [
-        b for b in bpps if not b.interchromosomal and b.break1.chr in dgv_regions_by_reference_name
-    ]:
-        for dgv_region in dgv_regions_by_reference_name[bpp.break1.chr]:
-            dist = abs(Interval.dist(Interval(dgv_region.start), bpp.break1))
-            if dist > lowest_resolution + distance:
-                break
-            elif (
-                dist > distance
-                or abs(Interval.dist(Interval(dgv_region.end), bpp.break2)) > distance
-            ):
-                continue
-            refname = dgv_region.reference_object
-            try:
-                refname = dgv_region.reference_object.name
-            except AttributeError:
-                pass
-            bpp.data['dgv'] = '{}({}:{}-{})'.format(
-                dgv_region.name, refname, dgv_region.start, dgv_region.end
-            )
+    bpps_copy = bpps.copy()
+    for variant in bpps:
+        variant.data['dgv'] = False
+        variant.data['known_sv_count'] = 0
+    dgv_flattened = list(chain(*dgv_regions_by_reference_name.values()))
```

Comment:
If we are going to flatten these anyway, maybe we shouldn't be reading them into a dict to begin with?

Reply:
Agreed, we should address this in V4, but to avoid breaking changes in the masking/dgv files  - we can re-use what the old code which requires the format to be `{(chr,chr): bpps}`